### ISSUE TITLE
Scrollbar interferes with other mouse events (drag content)

### ISSUE THREAD

**Issue by sebam91** [Grsmto/simplebar#240]
I am using additional plugin together with simplebar that allows you to directly drag scrollable content  with mouse (asvd/dragscroll). The problem is that dragging the simplebar scrollbar interferes with mouse movements which are bind on direct content mouse drag and work in opposite direction of scrollbar (like touch). Other way around works fine (mouse dragging content also moves the simplebar scrollbar).

Any suggestions? Does simplebar support any on scrollbar drag events so i can disable mouse dragging gestures specifically for the time when scrollbar is used?

**Comment by adjourn** [comment]
Would love to help but it's tricky to understand the issue and solve it without seeing it in action. Could you replicate your situation (i.e add drag plugin) to [this example](https://jsfiddle.net/19ta2s86/1/)?

**Comment by Grsmto** [comment]
SimpleBar is calling both `preventDefault` and `stopPropagation` on all these events: 'mousedown', `'click', 'dblclick', 'touchstart', 'touchend', 'touchmove'`. So I would have expect that your other plugin doesn't scroll when you drag the scrollbar. I'm not sure why it's happening as I would think the event does not get propagated. 

I looked at `asvd/dragscroll` and from what I see in the code, this should work fine with SimpleBar so I'm not sure why this is happening.

Are you using the latest version of SimpleBar (`simplebar@3.1.0-beta.5`)?

**Comment by sebam91** [comment]
I will prepare an example

**Comment by adjourn** [comment]
Here's a hack ([demo](https://jsfiddle.net/19ta2s86/4/)) in case this issue doesn't get resolved fast/need quick fix:

```js
const simplebar = new SimpleBar(document.getElementById('app'));
removePointerListeners();

function removePointerListeners() {
  ['mousedown', 'click', 'dblclick', 'touchstart', 'touchend', 'touchmove'].forEach((e) => {
    simplebar.el.removeEventListener(e, simplebar.onPointerEvent);
  });
}
```

It removes scrollbar/track interactions but it seems to maintains all other functionality.

**Comment by adjourn** [comment]
Here's a hack ([demo](https://jsfiddle.net/19ta2s86/4/)) in case this issue doesn't get resolved fast/need quick fix:

```js
const simplebar = new SimpleBar(document.getElementById('app'));
removePointerListeners();

function removePointerListeners() {
  ['mousedown', 'click', 'dblclick', 'touchstart', 'touchend', 'touchmove'].forEach((e) => {
    simplebar.el.removeEventListener(e, simplebar.onPointerEvent);
  });
}
```

It does remove scrollbar/track interactions but it seems to maintain all other functionality.

**Comment by adjourn** [comment]
Here's a hack ([demo](https://jsfiddle.net/19ta2s86/4/)) in case this issue doesn't get resolved fast/need quick fix:

```js
const simplebar = new SimpleBar(document.getElementById('app'));
removePointerListeners();

function removePointerListeners() {
  ['mousedown', 'click', 'dblclick', 'touchstart', 'touchend', 'touchmove'].forEach((e) => {
    simplebar.el.removeEventListener(e, simplebar.onPointerEvent);
  });
}
```

It does remove scrollbar/track interactions but it seems to maintain all other functionality.

PS! It could create issues I didn't think about, use at your own risk.

**Comment by sebam91** [comment]
Is there an option to access `simplebar-track` or `simplebar-scrollbar` directly? Not only `scrollElement` which detects the whole dragging area being scrolled. I'd like to keep functionality of both plugins, and go the other way, which is disabling the _dragscroll_ for the time of scrolling the content with `simplebar-scrollbar`.

I tried with

```
$('.simplebar-scrollbar').on('mousedown',function(){
// disable dragscroll
});
```

but the event is somehow not firing. Maybe i am doing something wrong.

**Comment by sebam91** [comment]
@asvd can you provide any additional information here?

**Comment by adjourn** [comment]
Track (and all descendants) has CSS rule `pointer-events: none;` which seems to be important for this implementation. It means that both are invisible for cursor and touch, therefore don't fire event you tried.

**Comment by adjourn** [comment]
Track (and all descendants) has CSS rule `pointer-events: none;` which seems to be important for this implementation. It means that both are invisible to cursor and touch, therefore don't fire event you tried.

**Comment by sebam91** [comment]
> Track (and all descendants) has CSS rule `pointer-events: none;` which seems to be important for this implementation. It means that both are invisible for cursor and touch, therefore don't fire event you tried.

Oh yeah, i totally missed that one. 😀 

It would be definitely a plus for `simplebar` to integrate an optional setting to allow mouse dragging the content. 😺

**Comment by Abed-Hasouneh** [comment]
@Grsmto Hey man !!
how I can get this fix for version 2.6.1 
I dont wont to change the version but i need to have the fix on it .
respictfully

**Comment by rcklmk** [comment]
Hi,

I know this issue is closed but I'm leaving this as a note to myself and hope somebody may find it useful as well.

I encountered a similar situation where I was trying to make simplebar work together with a calendar library I'm using and dragging the scrollbar causes unintended calendar operations right beneath after mouse release.

After inspecting the scrollbar, I found that `.simplebar-track` elements have `pointer-events: none` set which causes mouse events to fallthrough to the elements rendered below them. Overriding the style to `pointer-events: initial` fixed my problem and the solution can be less hacky if we avoid modifying `.simplebar-track` directly and apply the styles through the wrapper class selector of the simplebar container element like so:

```js
new SimpleBar(document.querySelector('.my-custom-scrollable-ui-wrapper'));
```

```css
.my-custom-scrollable-ui-wrapper .simplebar-track {
  pointer-events: initial;
}
```

I assume that `pointer-events: none` is set as default so that the scrollbar is treated as if it is not there in most situations, so there shouldn't be any big side effects. Also  `preventDefault` and `stopPropagation` don't work in this situation probably because `.simplebar-track` aren't children of the simplebar wrapped content:

```html
<div class="my-custom-scrollable-ui-wrapper">
  <div class="simplebar-wrapper"> ...  <!--  actual content here --> ... </div>
  <div class="simplebar-track simplebar-horizontal">...</div>
  <div class="simplebar-track simplebar-vertical">...</div>
</div>
```
